### PR TITLE
TF-867, TF-851,TF-501 Share image with TeamMail 

### PR DESCRIPTION
**Pull request by ManhNTX**
https://user-images.githubusercontent.com/107173849/187083613-b11658f9-d474-4751-b9ef-a46770caa884.mov

https://user-images.githubusercontent.com/107173849/187097621-807de9f6-4b67-4519-a9d6-55e788669e79.mov

https://user-images.githubusercontent.com/107173849/187334049-cd7b4130-5978-4918-b716-b314a1241afb.mov

https://user-images.githubusercontent.com/107173849/187400795-d9d1a0fe-d28c-463e-beab-87eaca5cf262.mov

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -176,7 +176,7 @@ dependencies:
   percent_indicator: 4.2.2
 
   # Isolate
-  worker_manager: 4.4.0
+  worker_manager: 4.4.6
```
Comment (dab246) [inline pubspec.yaml:179]: Why upgrade to version 4.4.6

File: lib/features/composer/presentation/composer_bindings.dart [hunk 2]
```diff
@@ -171,6 +171,7 @@ class ComposerBindings extends BaseBindings {
   void dispose() {
     Get.delete<UploadController>();
     Get.delete<RichTextWebController>();
+    Get.delete<RichTextMobileTabletController>();
```
Comment (dab246) [inline lib/features/composer/presentation/composer_bindings.dart:174]: why delete it?

File: lib/features/login/data/network/config/authorization_isolate_interceptors.dart [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:dio/dio.dart';
+import 'package:core/core.dart';
+import 'package:model/account/authentication_type.dart';
+import 'package:model/oidc/oidc_configuration.dart';
+import 'package:model/oidc/token.dart';
+
+class AuthorizationIsolateInterceptors extends QueuedInterceptorsWrapper {
```
Comment (dab246) [inline lib/features/login/data/network/config/authorization_isolate_interceptors.dart:10]: Why use `QueuedInterceptorsWrapper` and not `InterceptorsWrapper`

File: lib/features/mailbox/data/network/mailbox_isolate_worker.dart [hunk 4]
```diff
@@ -46,7 +45,7 @@ class MailboxIsolateWorker {
           totalEmailUnread,
           onProgressController);
     } else {
-      final result = await _isolateExecutor.execute(
+      final result = await Executor().execute(
```
Comment (dab246) [inline lib/features/mailbox/data/network/mailbox_isolate_worker.dart:48]: why not inject it and use it?

File: lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart [hunk 5]
```diff
@@ -123,17 +121,18 @@ class MailboxDashBoardController extends ReloadableController {
 
   @override
   void onInit() {
+    _registerProgressState();
     _registerNetworkConnectivityState();
-    _registerPendingEmailAddress();
```
Comment (dab246) [inline lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:127]: why remove it?

File: lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart [hunk 6]
```diff
@@ -123,17 +121,18 @@ class MailboxDashBoardController extends ReloadableController {
...
-    _registerPendingFileInfo();
```
Comment (dab246) [inline lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:128]: same

**Review by dab246 — COMMENTED** [review]
Why create more `intercepters` what does it do?

File: ios/TeamMailShareExtension/ShareViewController.swift [hunk 7]
```diff
@@ -5,7 +5,7 @@ import Photos
 
 class ShareViewController: SLComposeServiceViewController {
     var hostAppBundleIdentifier = ""
-    var appGroupId = ""
+    let appGroupId = "group.com.linagora.teammail"
```
Comment (hoangdat) [inline ios/TeamMailShareExtension/ShareViewController.swift:8]: No other way to make that work?

File: lib/features/mailbox/data/network/mailbox_isolate_worker.dart (hunk 4 above)
Comment (hoangdat) [reply lib/features/mailbox/data/network/mailbox_isolate_worker.dart:48]: How about?
```
final executor = Get.find......
```

File: lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart (hunk 6 above)
Comment (ManhNTX) [reply lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:128]: same

**Comment by hoangdat** [comment]
LGTM

File: lib/main.dart [hunk 8]
```diff
@@ -23,6 +24,7 @@ void main() async {
     ));
     await MainBindings().dependencies();
     await HiveCacheConfig().setUp();
+    await Executor().warmUp();
```
Comment (hoangdat) [inline lib/main.dart:27]: we need to check web or mobile to warmUp?

File: lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart [hunk 9]
```diff
@@ -647,7 +644,7 @@ class MailboxDashBoardController extends ReloadableController {
     _fileReceiveManagerStreamSubscription.cancel();
     _connectivityStreamSubscription.cancel();
     _progressStateController.close();
-    _isolateExecutor.dispose();
+    Executor().dispose();
```
Comment (hoangdat) [inline lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:647]: please inject instead

File: lib/main/bindings/main_bindings.dart [hunk 10]
```diff
@@ -11,6 +12,7 @@ class MainBindings extends Bindings {
     await CoreBindings().dependencies();
     LocalBindings().dependencies();
     NetworkBindings().dependencies();
+    NetworkIsolateBindings().dependencies();
```
Comment (hoangdat) [inline lib/main/bindings/main_bindings.dart:15]: have any impact to web version?

File: lib/main/bindings/network/network_isolate_binding.dart [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+  void dependencies() {
+    _bindingDio();
+    _bindingApi();
+    _bindingIsolateWorker();
+  }
+
+  void _bindingDio() {
+  final dio = Get.put(Dio(Get.find<BaseOptions>()), tag: BindingTag.isoLateTag);
+  Get.put(DioClient(dio), tag: BindingTag.isoLateTag);
+    _bindingInterceptors(dio);
```
Comment (hoangdat) [inline lib/main/bindings/network/network_isolate_binding.dart:29]: indent

File: lib/main.dart (hunk 8 above)
Comment (ManhNTX) [reply lib/main.dart:27]: checked

File: lib/main/bindings/main_bindings.dart (hunk 10 above)
Comment (ManhNTX) [reply lib/main/bindings/main_bindings.dart:15]: no impact

File: lib/main/bindings/network/network_isolate_binding.dart [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+  }
+
+  void _bindingInterceptors(Dio dio) {
+    Get.put(AuthorizationInterceptors(
+      dio,
+      Get.find<AuthenticationClientBase>(),
+      Get.find<TokenOidcCacheManager>(),
+      Get.find<AccountCacheManager>(),
+    ), tag: BindingTag.isoLateTag);
+    dio.interceptors.add(Get.find<DynamicUrlInterceptors>());
```
Comment (dab246) [inline lib/main/bindings/network/network_isolate_binding.dart:39]: Why not put `DynamicUrlInterceptors ` in `isolateTag`

Comment (ManhNTX) [reply lib/main/bindings/network/network_isolate_binding.dart:39]: DynamicUrlInterceptors already include network_binding and there is no need to create a new DynamicUrlInterceptors to use for dio of isolate. So I not put DynamicUrlInterceptors in isolateTag

Comment (hoangdat) [reply lib/main/bindings/network/network_isolate_binding.dart:39]: > no need to create a new DynamicUrlInterceptors to use for dio of isolate

I think it is different instance? Why?

**Review by dab246 — APPROVED** [review]
/lgtm

**Review by hoangdat — APPROVED** [review]

**Pull request merged**